Data/Config.hs

```diff
@@ -159,6 +161,49 @@ loadConfig path
              Left e    -> throwM e
              Right reg -> return $ Config reg
 
+--------------------------------------------------------------------------------
+-- | Loads a config without flattening any property object out.
+loadNotInlinedConfig :: (MonadIO m, MonadThrow m) => FilePath -> m Config
+loadNotInlinedConfig path
+    = do txt <- liftIO $ T.readFile path
+         pse <- parse path txt
+         let action
+                 = do ps         <- pse
+                      (tys, pts) <- typecheck $ rename ps
+                      return $ register tys pts
+         case action of
+             Left e    -> throwM e
+             Right reg -> return . Config . removeInlining . simplifyReg $ reg
+
+--------------------------------------------------------------------------------
+simplifyReg :: Reg -> Reg
+simplifyReg reg
+    = reg { regAST = fmap (simplify reg) (regAST reg) }
+
+--------------------------------------------------------------------------------
+removeInlining :: Reg -> Reg
+removeInlining reg
+    = reg { regAST = M.mapMaybeWithKey go (regAST reg) }
+  where
+    containsDot = T.any ('.' ==)
+
+    go key value
+        | containsDot key = Nothing
```

Comment:
This drops valid keys like

```hocon
another.string = "I'm a String"
```

I implemented this like so:

```haskell
    keep key = case T.break (== '.') key of
        (_, "") -> True                       -- keep, as it's a top-level key
        (l, _)  -> l `M.notMember` regAST reg -- keep if the first part isn't a top-level key

    go key value
        | keep key  = Just $ unscoped key value
        | otherwise = Nothing
```

Reply:
your example with my code will produce the following AST:

```
("another", OBJECT [Prop "string" "I'm a String"])
```

Which I thought was what you were looking for.

Reply:
So 2 things:

1. Initially I did think I wanted that AST (and I probably communicated that).  But later I realised that in JSON "another.string" is a valid key for a string, so I think it should stay together?  I haven't checked the HOCON spec if it says anything here.

2. I've rebased my ToJSON implementation and tests on this PR, and in that I see it dropping this key.  You can have a look at https://github.com/dwijnand/deiko-config/tree/to-aeson-rebased, running `cabal v2-test`